src/main/java/net/dries007/tfc/objects/te/TEPowderKeg.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.objects.te;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.inventory.InventoryHelper;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.ItemStackHandler;
+
+import net.dries007.tfc.objects.blocks.BlockLargeVessel;
+import net.dries007.tfc.objects.inventory.capability.IItemHandlerSidedCallback;
+import net.dries007.tfc.objects.inventory.capability.ItemHandlerSidedWrapper;
+import net.dries007.tfc.util.calendar.CalendarTFC;
+import net.dries007.tfc.util.calendar.ICalendarFormatted;
+
+import static net.dries007.tfc.objects.blocks.BlockLargeVessel.SEALED;
+
+/**
+ * @see BlockLargeVessel
+ */
+@ParametersAreNonnullByDefault
+public class TEPowderKeg extends TETickableInventory implements IItemHandlerSidedCallback
+{
+    private boolean sealed;
+    private long sealedTick, sealedCalendarTick;
```

Comment:
Why are either of these necessary? It's not used for performing recipes, this seems like copy pasta from large vessel.

Reply:
These report when the keg was sealed and allows the same code to be used to determine sealed state on serialize/deserialize. Thought it was a reasonable feature.

Reply:
We only need one of them though.